types/sinon/index.d.ts

```diff
@@ -233,9 +233,7 @@ declare namespace Sinon {
     }
 
     interface SinonFakeTimersStatic {
-        (): SinonFakeTimers;
-        (...timers: string[]): SinonFakeTimers;
-        (now: number, ...timers: string[]): SinonFakeTimers;
+        (now?: number | Date, config?: { now?: number | Date, toFake?: string[], shouldAdvanceTime?: boolean }): SinonFakeTimers;
```

Comment:
maybe here you could extract it into an interface:

```ts
interface SinonFakeTimersConfig {
  now: number | Date;
  toFake: string[];
  shouldAdvanceTime: boolean;
}
```

and have `config?: Partial<SinonFakeTimersConfig>` instead?

Reply:
Yes, defining an `interface` makes sense. See 947574d for updates.

I'm not sure it makes sense to reduce it further; doesn't feel like it gets easier to read or understand by combining it all into a single declaration. The way it is in 947574d is also closer to what they have defined in http://sinonjs.org/releases/v4.3.0/fake-timers/, imo.

Reply:
Fair enough. The guidelines on Typescript Do's and Don'ts are really concerned with trailing optional parameters and this case of multiple single parameter overloads isn't really the same thing. Getting rid of the ambiguous empty arg overload I think was the most important part.